airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/debezium/AirbyteDebeziumHandler.java

```diff
@@ -125,16 +127,18 @@ public AutoCloseableIterator<AirbyteMessage> getIncrementalIterators(final Debez
     final Long syncCheckpointRecords = config.has(SYNC_CHECKPOINT_RECORDS_PROPERTY)
         ? config.get(SYNC_CHECKPOINT_RECORDS_PROPERTY).asLong()
         : SYNC_CHECKPOINT_RECORDS;
-    return AutoCloseableIterators.fromIterator(new DebeziumStateDecoratingIterator<>(
-        eventIterator,
-        cdcStateHandler,
+
+    DebeziumMessageProducer messageProducer = new DebeziumMessageProducer(cdcStateHandler,
         targetPosition,
         eventConverter,
         offsetManager,
         trackSchemaHistory,
-        schemaHistoryManager.orElse(null),
-        syncCheckpointDuration,
-        syncCheckpointRecords));
+        schemaHistoryManager.orElse(null));
+
+    // Stream is not used in this iterator.
```

Comment:
nit: it's unclear what this comment refers to.

Reply:
fixed.